src/language/walker/blockScopeAwareRuleWalker.ts

```diff
@@ -58,6 +58,16 @@ export abstract class BlockScopeAwareRuleWalker<T, U> extends ScopeAwareRuleWalk
         return;
     }
 
+    public findBlockScope(predicate: (scope: U) => boolean) {
```

Comment:
why do we need this? shouldn't it just be `this.getAllBlockScopes().find(predicate)`?